upgrading/extending-daml-models/1_package_ids.md

```diff
@@ -0,0 +1,89 @@
+You've already learned how to extend a DAML model when you added the messaging feature to the mini
+social network of the [Getting Started
+Guide](https://daml.com/learn/getting-started/your-first-feature). 
+
+There's a catch though with extending a DAML model like this. The package ID of a DAML archive
+depends on the contents of the compiled DAML code. Changing the code results in a different package
+ID. This means that contracts created with the old package will be different from contracts created
+with the updated code and if you would just upload the new package to your ledger and replace the
+old one, you couldn't execute any choices on contracts anymore, that have been created with the
+original code.
+
+While this behaviour presents a challenge to a ledger operator, it is a blessing for your data
+integrity. It gives you a guarantee that the read and write rights to your data can not be changed
+in hindsight, and stay exactly as you defined it.
+
+Let's verify this.
+
+First, open `daml/User.daml`{{open}} in the IDE and **wait for it to load**. This is the familiar
+`User` module of the Getting Started Guide.
+
+In the terminal build the `create-daml-app` package with
+
+```
+daml build
+```{{execute T1}}
+
+Now inspect the package ID of the freshly package:
+
+```
+daml damlc inspect-dar .daml/dist/create-daml-app-0.1.0.dar
+```{{execute T1}}
+
+You will see a lengthy output listing the files and packages contained within your `create-daml-app`
+package. In the list of contained packages you see the main package:
```

Comment:
Maybe also add the list of/summary of what other packages users will be seeing in the output as it's quite long.

Reply:
that would be a really long output, I'll add the first 5 or so.